dags/deletes.py

```diff
@@ -412,20 +524,38 @@ def load_and_verify_deletes_dictionary(
     return dictionary
 
 
+@dagster.op
+def load_and_verify_adhoc_event_deletes_dictionary(
+    cluster: dagster.ResourceParam[ClickhouseCluster],
+    dictionary: AdhocEventDeletesDictionary,
+) -> AdhocEventDeletesDictionary:
+    """Load the dictionary data on all hosts in the cluster, and ensure all hosts have identical data."""
+    checksums = cluster.map_all_hosts(dictionary.load, concurrency=1).result()
+    assert len(set(checksums.values())) == 1
+    return dictionary
+
+
 @dagster.op
 def delete_events(
     context: dagster.OpExecutionContext,
     cluster: dagster.ResourceParam[ClickhouseCluster],
     load_and_verify_deletes_dictionary: PendingDeletesDictionary,
+    load_and_verify_adhoc_event_deletes_dictionary: AdhocEventDeletesDictionary,
 ) -> tuple[PendingDeletesDictionary, ShardMutations]:
     """Delete events from sharded_events table for pending deletions."""
 
     def count_pending_deletes(client: Client) -> int:
         result = client.execute(
             f"""
-            SELECT count()
-            FROM {load_and_verify_deletes_dictionary.qualified_name}
-            WHERE deletion_type IN ({DeletionType.Person}, {DeletionType.Team})
+            SELECT sum(pending)
+            FROM (
+                SELECT count() as pending
+                FROM {load_and_verify_deletes_dictionary.qualified_name}
+                WHERE deletion_type IN ({DeletionType.Person}, {DeletionType.Team})
+                UNION ALL
+                SELECT count() as pending
+                FROM {load_and_verify_adhoc_event_deletes_dictionary.qualified_name}
```

Comment:
Might be helpful to keep these two counts separate for reporting in metadata below instead of combining them together - not really a big thing but could see it being helpful to have a little bit more detail